src/prefect/orion/database/migrations/versions/postgresql/2023_01_13_045501_2882cd2df464_migrate_artifact_data.py

```diff
@@ -0,0 +1,125 @@
+"""Migrates state data to the artifact table
+
+Revision ID: 2882cd2df464
+Revises: 2882cd2df463
+Create Date: 2023-01-13 04:55:01.358638
+
+"""
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "2882cd2df464"
+down_revision = "2882cd2df463"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    ### START DATA MIGRATION
+
+    # insert nontrivial task run state results into the artifact table
+    def update_task_run_artifact_data_in_batches(batch_size, offset):
+        return f"""
+            INSERT INTO artifact (task_run_state_id, task_run_id, artifact_data)
+            SELECT id, task_run_id, _data
+            FROM task_run_state
+            WHERE _data != 'null' AND _data IS NOT NULL
```

Comment:
`_data` is not indexed, correct?

I think we'll have to get clever with our query, a temporary index, or maybe just visit all row by id?

Reply:
we already have temporary indexes on `artifact.task_run_state_id` and `artifact._flow_run_state_id` to improve the data migration, we can add one here as well and remove it in the cleanup migration

Reply:
A temporary index on `data` has been added into the `implement_artifact_table` schema migration and then removed in the `cleanup_artifact_migration` versions.